plugins/kinetic-dp/fu-kinetic-dp-secure-aux-isp.c

```diff
@@ -0,0 +1,853 @@
+							10000, 200, error)) {
+		g_prefix_error (error, "Sending App... failed!");
+		return FALSE;
+	}
+
+	g_message ("Sending App... done!");
+
+	// Send App initialized data
+	g_message ("Sending App init data... started!");
+
+	ptr = (guint8 *)fw_data + (fu_kinetic_dp_firmware_get_is_fw_esm_xip_enabled (firmware) ? SPI_APP_EXTEND_INIT_DATA_START : SPI_APP_NORMAL_INIT_DATA_START);
+	if (!fu_kinetic_dp_secure_aux_isp_send_payload (connection, ptr, fu_kinetic_dp_firmware_get_app_init_data_size (firmware),
+							10000, 200, error)) {
+		g_prefix_error (error, "Sending App init data... failed!");
+		return FALSE;
+	}
+
+	g_message ("Sending App init data... done!");
+
+	if (fu_kinetic_dp_firmware_get_cmdb_block_size (firmware)) {
+		// Send CMDB
+		g_message ("Sending CMDB... started!");
+
+		ptr = (guint8 *)fw_data + SPI_CMDB_BLOCK_START;
+		if (!fu_kinetic_dp_secure_aux_isp_send_payload (connection, ptr, fu_kinetic_dp_firmware_get_cmdb_block_size (firmware),
+								10000, 200, error)) {
+			g_prefix_error (error, "Sending CMDB... failed!");
+			return FALSE;
+		}
+
+		g_message ("Sending CMDB... done!");
+	}
+
+	// Send Application Identifier
+	ptr = (guint8 *)fw_data + SPI_APP_ID_DATA_START;
+	if (!fu_kinetic_dp_secure_aux_isp_send_payload (connection, ptr, STD_APP_ID_SIZE, 10000, 200, error)) {
+		g_prefix_error (error, "Sending App ID data... failed!");
+		return FALSE;
+	}
+
+	g_message ("Sending App ID data... done!");
+
+	return TRUE;
+}
+
+static gboolean
+fu_kinetic_dp_secure_aux_isp_install_fw_images (FuKineticDpConnection *connection,
+						GError **error)
+{
+	guint8 cmd_id = KT_DPCD_CMD_INSTALL_IMAGES;
+	guint8 status;
+	guint16 wait_count = 1500;
+	guint16 progress_inc = FLASH_PROGRAM_COUNT / ((read_flash_prog_time * 1000) / WAIT_PROG_INTERVAL_MS);
+
+	g_message ("Installing F/W payload... started");
+
+	if (!fu_kinetic_dp_secure_aux_isp_write_kt_prop_cmd (connection, cmd_id, error))
+	{
+		g_prefix_error (error, "Sending DPCD command... failed!");
+		return FALSE;
+	}
+
+	while (wait_count-- > 0) {
+		if (!fu_kinetic_dp_connection_read (connection, DPCD_ADDR_FLOAT_CMD_STATUS_REG, &status, 1, error)) {
+			g_prefix_error(error, "Reading DPCD_MCA_CMD_REG... failed!");
+			return FALSE;
+		}
+
+		if (status != (cmd_id | DPCD_KT_CONFIRMATION_BIT)) {	// Target response
+			if (status == cmd_id) {				// Confirmation bit if cleared
+				isp_payload_procd_size += (isp_total_data_size - isp_procd_size);
+				g_message ("Programming F/W payload... done!");
+
+				return TRUE;
+			} else {
+				g_prefix_error (error, "Installing images... failed!");
+
+				return FALSE;
+			}
+		}
+
+		if (isp_procd_size < isp_total_data_size) {
+			isp_procd_size += progress_inc;
+			isp_payload_procd_size += progress_inc;
+		}
+
+		// Wait 50ms
+		g_usleep (50 * 1000);
+	}
+
+	g_prefix_error (error, "Installing images... timed-out!");
```

Comment:
don't end this line with` !`, it should be: `timed-out: ` so the prefix looks correct on the error.

Reply:
And actually this should probably be a `g_set_error`, error never got set.